### DISCUSSION TITLE
Cannot input newlines into multi-line text fields

### DISCUSSION THREAD

**Discussion by xylix** [flet-dev/flet#1224] [Q&A] [answered]
**Description**

When editing a multi-line textfield and pressing the return key or shift+return, I would expect a newline to be input. Instead focus is moved away from the input,

**Code example to reproduce the issue:**

```python
import flet as ft
from flet_core import TextField
def main(page):
    textfield = TextField(min_lines=2, max_lines=4)
    page.add(
        ft.Text("Multiline text input"),
        textfield
    )
ft.app(target=main)
```

**Describe the results you received:**

When editing a multi-line textfield and pressing the return key or shift+return, I would expect a newline to be input. Instead focus is moved away from the input,

**Describe the results you expected:**

Ability to enter newlines into multi-line textfields with return or shift-return.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:

```
Name: flet
Version: 0.4.2
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page: 
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: /Users/kerkko/Code/tmp/flet-textfield-newline-test/venv/lib/python3.11/site-packages
Requires: flet-core, httpx, oauthlib, packaging, watchdog, websocket-client, websockets
Required-by: 
```

**Operating system**:

macOS monterey 12.6 21G115

**Additional environment details:**

**Comment by burque505** [comment]
Needs 'multiline=True'. Try this:
```
import flet as ft
from flet_core import TextField
def main(page):
    textfield = TextField(multiline=True, min_lines=2, max_lines=4)
    page.add(
        ft.Text("Multiline text input"),
        textfield
    )
ft.app(target=main)
```

**Comment by gbrandt** [comment]
Is this identical to #1106 ?

**Comment by FeodorFitsner** [comment, marked as answer]
@gbrandt I believe `multiline=True` is the solution.
